usdigitalresponse/usdr-gost
# Search panel - Apply functionality

### Ticket #1403 
Encompasses 
#1403  
#1404
#1405 
#1406 

## Description
Basic applying of advanced search to grants table logic for Grants and My Grants views. Includes:
* Adding of fields in form to the search, **where the API supports it as far as I could tell**
  * A number of fields do not support this - their filter chips will display with a ⚠️ 
* Filter chips display and can be cleared, which will reflect in the search
* Table heading filters are removed from the new table to match the [design](https://www.figma.com/file/TRONzuKDV3K9gzWZAwJ7ma/Search-Experience?node-id=194%3A17678&mode=dev)  


<details>
  <summary>Draft description for posterity's sake</summary>
Still a lot of ground to cover here, but I wanted to get a draft/WIP up before I get too much further. New to Vue so please stop me now if I'm going about this the wrong way! 

Basic functionality:
1. Applying _some_ fields from the form works. Not all are hooked up, and some aren't in the API yet. This includes include/exclude, likely others. I've mostly been testing on categories and cost sharing so far.
2. Filter chips should display correctly and be clearable
3. I went ahead and removed the old column filters and search input per the [design](https://www.figma.com/file/TRONzuKDV3K9gzWZAwJ7ma/Search-Experience?node-id=194%3A17678&mode=dev), lmk if that's not right.

Known issues:
1. As mentioned above, some fields aren't hooked up for search. Some can be for this PR, but some (like include) will require API changes.
2. Because there's one state module, My Grants and Grants will retain filters between the pages. Need to bifurcate that or clear upon route navigation.
4. Form state is a little weird wrt how it stays when you re-open
</details>


## Screenshots / Demo Video

https://github.com/usdigitalresponse/usdr-gost/assets/624510/3f4231fe-5dda-4b96-941c-67275495880e


## Testing
1. Run locally, make sure `VUE_APP_USE_NEW_TABLE=true` is set in `client/.env`. Note that this value won't hot reload.
2. Navigate to Browse Grants page
3. Click Search button. Side panel should open.
4. Enter values, hit Apply. 
4a.  It can help to edit some values in `packages/server/seeds/dev/ref/grants.js` and re-run `yarn db:seed` so you have more variety to search for.
6. Click the "X" on chips, the filters should update (providing they aren't ⚠️ fields).
7. Click "Clear All". All filters should clear.
8. Navigate to My Grants.
9. Click the My Grants view.
10. Repeat the above with the various Interested, Assigned, Not Applying, and Applied filters.  
10a. **Note** There is functional overlap between the review status filters and the "default" filters on the various My Grants tabs. Based on the designs, I think those tabs should be removed - but I'm not familiar enough with My Grants and how review statuses work to say for certain.

### Automated and Unit Tests
- [ ] Added Unit tests (🔴 **Any guidance here**? I didn't find existing Grants tests, only ARPA.)

### Manual tests for Reviewer

## Checklist
